tapchannelmsg/wire_msgs.go

```diff
@@ -0,0 +1,354 @@
+func (t *TxAssetOutputProof) Amt() fn.Option[uint64] {
+	return fn.Some(t.AssetOutput.Val.Amount)
+}
+
+// A compile time check to ensure TxAssetOutputProof implements the
+// AssetFundingMsg interface.
+var _ AssetFundingMsg = (*TxAssetOutputProof)(nil)
+
+// AssetFundingCreated is sent by the initiator of the funding flow after
+// they've able to fully finalize the funding transaction. This message will be
+// sent before the normal funding_created message.
+type AssetFundingCreated struct {
+	// TempChanID is the temporary channel ID that was assigned to the
+	// channel.
+	TempChanID tlv.RecordT[tlv.TlvType0, funding.PendingChanID]
+
+	// FundingOutputs are the completed set of funding output proofs. The
+	// remote party will use the transition (suffix) proofs encoded in the
+	// funding output to be able to create the aux funding+commitment
+	// blobs.
+	FundingOutputs tlv.RecordT[tlv.TlvType1, AssetOutputListRecord]
+}
+
+// NewAssetFundingCreated creates a new AssetFundingCreated message.
+func NewAssetFundingCreated(pid funding.PendingChanID,
+	outputs []*AssetOutput) *AssetFundingCreated {
+
+	return &AssetFundingCreated{
+		TempChanID: tlv.NewPrimitiveRecord[tlv.TlvType0](pid),
+		FundingOutputs: tlv.NewRecordT[tlv.TlvType1](
+			AssetOutputListRecord{
+				Outputs: outputs,
+			},
+		),
+	}
+}
+
+// MsgType returns the type of the message.
+func (t *AssetFundingCreated) MsgType() lnwire.MessageType {
+	return AssetFundingCreatedType
+}
+
+// Decode reads the bytes stream and converts it to the object.
+func (t *AssetFundingCreated) Decode(r io.Reader, _ uint32) error {
+	stream, err := tlv.NewStream(
+		t.TempChanID.Record(),
+		t.FundingOutputs.Record(),
+	)
+	if err != nil {
+		return err
+	}
+
+	return stream.Decode(r)
+}
+
+// Encode converts object to the bytes stream and write it into the write
+// buffer.
+func (t *AssetFundingCreated) Encode(w *bytes.Buffer, _ uint32) error {
+	stream, err := tlv.NewStream(
+		t.TempChanID.Record(),
+		t.FundingOutputs.Record(),
+	)
+	if err != nil {
+		return err
+	}
+
+	return stream.Encode(w)
+}
+
+// PendingChanID returns the temporary channel ID that was assigned to the
+// channel.
+func (t *AssetFundingCreated) PendingChanID() funding.PendingChanID {
+	return t.TempChanID.Val
+}
+
+// Amt returns the amount of the asset that this output represents.
+func (t *AssetFundingCreated) Amt() fn.Option[uint64] {
+	return fn.Some(t.FundingOutputs.Val.Sum())
+}
+
+// AssetFundingAck is sent by the responder of the funding flow after they've
+// received the funding input and output proofs. If the responder is able to
+// fully validate the proposed funding parameters, then they'll send this method
+// with accept=true. Otherwise, they'll send with accept set to false.
+type AssetFundingAck struct {
+	// TempChanID is the temporary channel ID that was assigned to the
+	// channel.
+	TempChanID tlv.RecordT[tlv.TlvType0, funding.PendingChanID]
```

Comment:
is this always exactly the "pending" channel ID? If so can we just name it `PendingChanID` for consistency?

Reply:
I think it is, yes. Renamed :+1: